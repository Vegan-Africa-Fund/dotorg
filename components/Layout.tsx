import React, { ReactNode } from 'react'
import Link from 'next/link'
import Head from 'next/head'
import { makeStyles } from '@material-ui/core/styles';
import { Box, Menu, IconButton, MenuItem} from '@material-ui/core'
import MenuIcon from '@material-ui/icons/Menu';
import Footer from './Footer';

type Props = {
  children?: ReactNode
  title?: string
}

const useStyles = makeStyles(theme => ({
  nav: {
    display: "flex",
    width: "100%",
    justifyContent: "space-around",
    [theme.breakpoints.down('sm')] : {
      display: "none"
    },
    '& h4': {
      borderLeft: "1px solid black",
      width: "25%",
      textAlign: "center",
      fontSize: "1.2rem",
      fontWeight: "400"
    },
    '& h4:first-child': {
      borderLeft: "0"
    }
  }
}))


const Layout = ({ children, title = 'This is the default title' }: Props) => {
  const classes = useStyles();

  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);

  const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = () => {
    setAnchorEl(null);
  };
  return (
  <>
    <Head>
      <title>{title}</title>
      <meta charSet="utf-8" />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
    </Head>
    <header>
      <nav className= {classes.nav}>
        <Link href="#"><h4>Current Projects</h4></Link>
        <Link href="#"><h4>Blog</h4></Link>
        <Link href="#"><h4>Proposed Projects</h4></Link>
        <Link href="#"><h4>Donate</h4></Link>
      </nav>
      <Box display={{ xs: 'block', md: 'none' }}>
      <IconButton aria-controls="simple-menu" aria-haspopup="true" onClick={handleClick}>
        <MenuIcon fontSize="large"/>
      </IconButton>
      <Menu
        id="simple-menu"
        anchorEl={anchorEl}
        keepMounted
        open={Boolean(anchorEl)}
        onClose={handleClose}
      >
        <MenuItem onClick={handleClose}>Current Projects</MenuItem>
        <MenuItem onClick={handleClose}>Blog</MenuItem>
        <MenuItem onClick={handleClose}>Proposed Projects</MenuItem>
        <MenuItem onClick={handleClose}>Donate</MenuItem>
      </Menu>
      </Box>
    </header>
    <Box
    width="75%"
    mx="auto"
    lineHeight={4}
    mb="4rem"
    >
    {children}
    </Box>
    <footer>
      <Footer />
    </footer>
  </>
)}

export default Layout
