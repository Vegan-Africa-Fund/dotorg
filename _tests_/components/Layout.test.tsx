export {}
import React from 'react';
import renderer from 'react-test-renderer';
import Layout from '../../components/Layout';

test("Layout is rendered correctly", () => {
 const layout = renderer.create(<Layout />).toJSON()
 expect(layout).toMatchSnapshot()
})