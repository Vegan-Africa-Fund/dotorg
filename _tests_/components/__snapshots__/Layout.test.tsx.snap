// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout is rendered correctly 1`] = `
Array [
  <header>
    <nav>
      <a
        aria-disabled={false}
        className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-outlined MuiButton-outlinedPrimary"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
      >
        <span
          className="MuiButton-label"
        >
          home
        </span>
      </a>
    </nav>
  </header>,
  <div
    className="MuiBox-root MuiBox-root-2"
  />,
  <footer>
    <header
      className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorDefault makeStyles-appBar-3 mui-fixed MuiPaper-elevation4"
    >
      <div
        className="MuiBox-root MuiBox-root-5"
      >
        <div
          className="MuiBox-root MuiBox-root-6"
        >
          <p
            className="MuiTypography-root MuiTypography-body1"
          >
            Fund:
            <span
              className="makeStyles-span-4"
            >
              ₿0.71522656
            </span>
             
          </p>
        </div>
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          <span
            className="makeStyles-span-4"
          >
            bitcoin
          </span>
          mission©2021
        </p>
      </div>
    </header>
  </footer>,
]
`;
